mountpoint-s3-fs/src/fs.rs

```diff
@@ -147,19 +145,11 @@ where
         client: Client,
         prefetch_builder: PrefetcherBuilder<Client>,
         runtime: Runtime,
-        bucket: &str,
-        prefix: &Prefix,
+        metablock: Box<dyn Metablock>,
```

Comment:
Here and above, if possible, take a `impl Metablock` and box (or rather "arc") it when setting the field.